l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -521,10 +514,12 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         require(!_isEraLegacyDeposit(_chainId, _l2BatchNumber, _l2TxNumberInBatch), "ShB: legacy cFD");
         {
             bytes32 dataHash = depositHappened[_chainId][_l2TxHash];
-            address l1Token = nativeTokenVault.tokenAddress(_assetId);
-            (uint256 amount, address prevMsgSender) = abi.decode(_assetData, (uint256, address));
-            bytes32 txDataHash = keccak256(abi.encode(prevMsgSender, l1Token, amount));
-            require(dataHash == txDataHash, "ShB: d.it not hap");
+            // Checking, if it was the legacy deposit, check new encoding otherwise
+            bool isLegacyTxDataHash = _isLegacyTxDataHash(_depositSender, _assetId, _assetData, dataHash);
+            if (!isLegacyTxDataHash) {
+                bytes32 txDataHash = this._encodeTxDataHash(false, _depositSender, _assetId, _assetData);
+                require(dataHash == txDataHash, "ShB: d.it not hap");
```

Comment:
we should check for legacyTxDataHash == true case as well, no?

Reply:
why?
We're passing the expected dataHash to `_isLegacyTxDataHash` and return true only if it is the case

Reply:
let's add a comment about it, it is indeed not noticable on the first glance